cpu/sam0_common/periph/timer.c

```diff
@@ -332,20 +336,24 @@ int timer_clear(tim_t tim, int channel)
 unsigned int timer_read(tim_t tim)
 {
     /* WORKAROUND to prevent being stuck there if timer not init */
-    if (!dev(tim)->CTRLA.bit.ENABLE) {
+    if (!(dev(tim)->CTRLA.reg & TC_CTRLA_ENABLE)) {
         return 0;
     }
 
     /* request synchronisation */
 #ifdef TC_CTRLBSET_CMD_READSYNC_Val
+    uint32_t cmd;
     dev(tim)->CTRLBSET.reg = TC_CTRLBSET_CMD_READSYNC;
      /* work around a possible hardware bug where it takes some
         cycles for the timer peripheral to set the SYNCBUSY/READSYNC bit
         after writing the READSYNC bit
 
         The problem was observed on SAME54.
       */
-    while(dev(tim)->CTRLBSET.bit.CMD == TC_CTRLBSET_CMD_READSYNC_Val) {}
+    do {
+        cmd = ((dev(tim)->CTRLBSET.reg & TC_CTRLBSET_CMD_Msk) >> TC_CTRLBSET_CMD_Pos);
+    } while(cmd == TC_CTRLBSET_CMD_READSYNC_Val);
+
```

Comment:
might be faster but please ignore unless someone objdumped

```suggestion
    do {
        cmd = ((dev(tim)->CTRLBSET.reg & TC_CTRLBSET_CMD_Msk));
    } while(cmd == TC_CTRLBSET_CMD_READSYNC_Val << TC_CTRLBSET_CMD_Pos);

```